tensorboard/plugins/projector/vz_projector/vz-projector-inspector-panel.ts

```diff
@@ -282,12 +339,36 @@ namespace vz_projector {
       this.searchBox.message = '';
       const minDist = neighbors.length > 0 ? neighbors[0].dist : 0;
 
+      if (this.spriteImagesAvailable) {
+        var spriteImagePath = this.spriteMeta.imagePath;
+        var {aspectRatio, nCols} = this.spriteMeta;
+        var paddingBottom = 100 / aspectRatio + '%';
+        var backgroundSize = `${nCols * 100}% ${nCols * 100}%`;
+        var backgroundImage = `url(${CSS.escape(spriteImagePath)})`;
+      }
+
       for (let i = 0; i < neighbors.length; i++) {
         const neighbor = neighbors[i];
 
         const neighborElement = document.createElement('div');
         neighborElement.className = 'neighbor';
 
+        if (this.spriteImagesAvailable) {
+          var neighborElementImage = document.createElement('div');
+          neighborElementImage.className = 'neighbor-image';
+          neighborElementImage.style.backgroundImage = backgroundImage;
+          neighborElementImage.style.paddingBottom = paddingBottom;
+          neighborElementImage.style.backgroundSize = backgroundSize;
+          const [row, col] = [
+            Math.floor(neighbor.index / nCols),
+            neighbor.index % nCols,
+          ];
+          neighborElementImage.style.backgroundPosition = `${(col /
+            (nCols - 1)) *
+            100}%
```

Comment:
yeah that weird thing was done by "prettier" .....

Reply:
Yep, sorry. :-) It’s mostly very good, but it has a hard time with
template strings because it can’t add any whitespace in the middle of
them (because they’re actually strings), and it doesn’t do any AST
transformations (`` `foobar` `` to `` `foo` + `bar` ``, e.g.) so
sometimes with template strings specifically you have to help it out to
get the result to actually be pretty.

Reply:
god prettier is so annoying..